worker-batch/src/main/java/com/hpe/caf/worker/batch/BatchWorker.java

```diff
@@ -72,23 +74,20 @@ public WorkerResponse doWork() throws InterruptedException, TaskRejectedExceptio
                 batchWorkerPlugin = (BatchWorkerPlugin) pluginClass.newInstance();
             }
             batchWorkerPlugin.processBatch(batchWorkerServices, task.batchDefinition, task.taskMessageType, task.taskMessageParams);
-            if (!messagePublisher.isTaskMessageBufferEmpty()) {
-                messagePublisher.publishLastMessage();
-            }
 
-            BatchWorkerResult result = new BatchWorkerResult();
-            result.batchTask = tracking == null ? task.targetPipe : tracking.getJobId();
+            final BatchWorkerResult result = new BatchWorkerResult();
+            result.batchTask = jobId;
 
             // Read configuration entry for return value behaviour
             if(configuration.getReturnValueBehaviour()==null) {
-                return createSuccessResult(result);
+                return createSuccessAndCompleteResponse(result);
             }
 
             switch(configuration.getReturnValueBehaviour())
             {
                 case RETURN_ALL:
                     // We return all results to the output queue
-                    return createSuccessResult(result);
+                    return createSuccessAndCompleteResponse(result);
                 case RETURN_NONE:
                     //  Return nothing to the output queue.
                     if (batchWorkerServices.hasSubtasks()) {
```

Comment:
Isn't this check now redundant?  Doesn't it need to use `return createTaskCompleteResponse();` regardless of whether there are subtasks are not?

Reply:
Doesn't it need to use return createTaskCompleteResponse(); regardless of whether there are subtasks are not?